doc/libpmemobj/pmemobj_tx_begin.3.md

```diff
@@ -315,6 +320,68 @@ The **TX_END** macro cleans up and closes the transaction started by the
 It is mandatory to terminate each transaction with this macro. If the transaction
 was aborted, *errno* is set appropriately.
 
+## TUNING ##
+
+From pmemobj implementation perspective there are two types of operations
+in a transaction:
+
++ **snapshots**, where action must be performed and persisted immediately
++ **intents**, where the persist part of the action can be postponed until
+  transaction commit
+
+**pmemobj_tx_add_range**() and all its variants belong to the **snapshots**
+group.
+
+**pmemobj_tx_alloc**() (with its variants), **pmemobj_tx_free**(),
+**pmemobj_tx_realloc**() (with its variants) and **pmemobj_tx_publish**()
+belong to the **intents** group. Even though **pmemobj_tx_alloc**() allocates
+memory immediately, it modifies only the runtime state and postpones persistent
+memory modifications to the commit phase. **pmemobj_tx_free**() cannot free
+the object immediately, because of possible transaction rollback, so it
+postpones both the action and persistent memory modifications to the commit
+phase. **pmemobj_tx_realloc**() is just a combination of those two and
+**pmemobj_tx_publish**() postpones reservations and deferred frees to
+the commit phase.
+
+Those two types of operations require that pmemobj builds a persistent log
+of operations. Intent log is applied on commit (sometimes it's called
+a redo log) and snapshot log is applied on abort (again, sometimes it's
+called an undo log).
+
+When pmemobj transactions starts it's not possible to predict how much
+persistent memory space will be needed for those logs. This means that pmemobj
+must internally allocate this space whenever it's needed. This has 2 down sides:
+
++ when transaction makes a lot of snapshots or does a lot of allocations,
+  pmemobj may need to do many internal allocations, which must be freed when
+  transaction ends, adding a lot of overhead when such big transactions are
```

Comment:
"adding a lot of overhead" 
 overhead of what? use more specific description or rm this at all.

"such big transactions"
hm, at first you said "a lot of snapshots or..." 
As a user I try to understand whether the problem is a lot of allocations or large allocations and when the feature will be useful for me.